runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -154,25 +171,40 @@ public Layout prepare(long epoch, long rank, Layout layout)
     @SuppressWarnings("unchecked")
     public Layout propose(long epoch, long rank, Layout layout)
             throws QuorumUnreachableException, OutrankedException {
-        CompletableFuture<Boolean>[] proposeList = layout.getLayoutClientStream()
-                .map(x -> x.propose(epoch, rank, layout))
+        CompletableFuture<Boolean>[] proposeList = getLayout().getLayoutServers().stream()
+                .map(x -> {
+                    CompletableFuture<Boolean> cf = new CompletableFuture<>();
+                    try {
+                        // Connection to router can cause network exception too.
+                        LayoutClient layoutClient = runtime.getRouter(x)
+                                .getClient(LayoutClient.class);
+                        cf =  layoutClient.propose(epoch, rank, layout);
+                    } catch (NetworkException e) {
```

Comment:
Do u need both **TimeoutException | NetworkException ** ?  Also, would you want to be pessimistic and catch all exceptions ?

Reply:
Only Network is expected here as that can be thrown on router creation.
Timeout is only expected as a result of the future completing exceptionally.

We can catch all exceptions later below when waiting for some future to complete, but only the 3 caught are expected there (Network, Timeout and WrongEpoch). Anything else would be unexpected and would be better if they are caught by the above layers (FailureHandler).